src/nacl/bindings/crypto_shorthash.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2016 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+from nacl._sodium import ffi, lib
+
+BYTES = lib.crypto_shorthash_siphash24_bytes()
+KEYBYTES = lib.crypto_shorthash_siphash24_keybytes()
+
+
+def crypto_shorthash_siphash24(data, key=b''):
+    """Compute a fast, cryptographic quality, keyed hash of the input data
+
+    :param data:
+    :type data: bytes
+    :param key:
+    :type key: bytes
+    """
+    _key = ffi.new("unsigned char[]", KEYBYTES)
+    digest = ffi.new("unsigned char[]", BYTES)
+    ffi.memmove(_key, key, min(len(key), KEYBYTES))
+    # The key will be truncated or zero-padded to KEYBYTES length
```

Comment:
I've just been following what I'd done for blake2b, whose key parameter is zero padded.